homeassistant/components/openai_conversation/conversation.py

```diff
@@ -173,27 +183,10 @@ async def _async_call_api(
                 for tool in session.llm_api.tools
             ]
 
-        messages: list[ChatCompletionMessageParam] = []
-        for message in session.async_get_messages(user_input.agent_id):
-            if message.native is not None and message.agent_id == user_input.agent_id:
-                messages.append(message.native)
-            else:
-                messages.append(
-                    cast(
-                        ChatCompletionMessageParam,
-                        {"role": message.role, "content": message.content},
-                    )
-                )
-
-        LOGGER.debug("Prompt: %s", messages)
-        LOGGER.debug("Tools: %s", tools)
-        trace.async_conversation_trace_append(
-            trace.ConversationTraceEventType.AGENT_DETAIL,
-            {
-                "messages": session.messages,
-                "tools": session.llm_api.tools if session.llm_api else None,
-            },
-        )
+        messages = [
+            _chat_message_convert(message)
+            for message in session.async_get_agent_messages()
```

Comment:
Doesn't this change break continuous conversation across assistants? It doesn't include other agents messages in the history, including [the fallback agent](https://github.com/home-assistant/core/blob/dev/homeassistant/components/assist_pipeline/pipeline.py#L1104-L1109). We should only filter out the native messages, but any interaction between other assistants and the user in the same chat history should still be considered

Reply:
<img width="414" alt="image" src="https://github.com/user-attachments/assets/aeb43596-c30c-43d3-9e1a-2d47d33ee963" />

with the new code it thinks the user wrote 2 messages, as it doesn't see that the first one was already handled.

Reply:
Ok I've been pretty confused about what is going on here and how this API is supposed to work. I think I may be getting closer to understanding. I didn't understand this was about fallback.

Reply:
could we not store native chat messages in the history? It seems like it's for an optimization? Which I'm not sure is needed.

Reply:
Yeah that's fine too. Not sure if needed either.

Reply:
Reverting these changes to match the existing logic.